### PR TITLE
Added removing path recursively.

### PR DESCRIPTION
**Pull request by mocarz**

### CODE REVIEW

File: src/ios-deploy/ios-deploy.m [hunk 1]
```diff
@@ -1436,6 +1438,59 @@ void remove_path(AMDeviceRef device) {
...
+    int ret = AFCRemovePath(conn, path);
+    if(ret == 0 || ret == 8) {
+        // Successfully removed (it was empty) or does not exist
+        NSLogVerbose(@"Deleted %s", path);
+        return;
+    }
+    
+    iter_dir(conn, path, (iter_callback) remove_path_callback);
+    
+    assert(AFCRemovePath(conn, path) == 0);
```
Comment (AntonShpk) [inline src/ios-deploy/ios-deploy.m]: it make possible to make empty Documents folder without assert:

    iter_dir(conn, path, (iter_callback) remove_path_callback);
    ret = AFCRemovePath(conn, path);
    if (ret == 10)
    {
        // No permissions for deleting folder
        NSLogVerbose(@"No permissions for delete %s", path);
        return;
    }

Comment (shazron) [inline src/ios-deploy/ios-deploy.m]: Not sure why you are commenting on your own code -- did you want to update it?

Comment (shazron) [inline src/ios-deploy/ios-deploy.m]: Sorry I got mixed up on who the author was here :)

Comment (shazron) [inline src/ios-deploy/ios-deploy.m]: @mocarz did you want to update the code based on @AntonShpk 's suggestion? Trying to get 1.9.2 out, I could get this in.

**Comment by shazron** [comment]
if this is still relevant, update it according to the comments above, then re-open it. Thank you!

**Pull request closed without merging**